kernel.py

```diff
@@ -228,7 +251,7 @@ def _thread_loop(self):
         loop.add_callback(self._start_kernel)
         try:
             loop.start()
-        except KeyboardInterrupt:
+        except (KeyboardInterrupt, SystemExit):
```

Comment:
Why? In what cases will a `SystemExit` raised here? And if it is raised, then shouldn't it actually handle it?

Reply:
Is this related to `keepkernel_on_exit`?

Reply:
Short story: It is obsolete, from a previous implementation I did not commit

I did not manage not trigger it. I could not trigger the KeyboardInterrupt either.
Yes it is for exiting the kernel, with a `sys.exit` <- `exit(0)`.
I did not see the `atexit.register` so I wanted to clean the stdio redirection there.

Reply:
572d8f5